src/osemgrep/reporting/Sarif_output.ml

```diff
@@ -0,0 +1,357 @@
+      ]
+
+let results (cli_output : OutT.cli_output) =
+  let result (cli_match : OutT.cli_match) =
+    let location =
+      `Assoc
+        [
+          ( "physicalLocation",
+            `Assoc
+              [
+                ( "artifactLocation",
+                  `Assoc
+                    [
+                      ("uri", `String (Fpath.to_string cli_match.path));
+                      ("uriBaseId", `String "%SRCROOT%");
+                    ] );
+                ( "region",
+                  `Assoc
+                    [
+                      ( "snippet",
+                        `Assoc [ ("text", `String cli_match.extra.lines) ] );
+                      ("startLine", `Int cli_match.start.line);
+                      ("startColumn", `Int cli_match.start.col);
+                      ("endLine", `Int cli_match.end_.line);
+                      ("endColumn", `Int cli_match.end_.col);
+                    ] );
+              ] );
+        ]
+    in
+    let suppression =
+      match cli_match.extra.is_ignored with
+      | None
+      | Some false ->
+          []
+      | Some true ->
+          [
+            ("suppressions", `List [ `Assoc [ ("kind", `String "inSource") ] ]);
+          ]
+    in
+    let fix = sarif_fix cli_match in
+    `Assoc
+      ([
+         ("ruleId", `String (Rule_ID.to_string cli_match.check_id));
+         ("message", `Assoc [ ("text", `String cli_match.extra.message) ]);
+         ("locations", `List [ location ]);
+         ( "fingerprints",
+           `Assoc [ ("matchBasedId/v1", `String cli_match.extra.fingerprint) ]
+         );
+         ("properties", `Assoc []);
+       ]
+      @ suppression @ fix)
+  in
+  List_.map result cli_output.results
+
+let error_to_sarif_notification (e : OutT.cli_error) =
+  let level = sarif_severity_of_severity e.level in
+  let message =
+    Option.value
+      ~default:
+        (Option.value
+           ~default:(Option.value ~default:"" e.short_msg)
+           e.long_msg)
+      e.message
+  in
+  let descriptor = Error.string_of_error_type e.type_ in
+  `Assoc
+    [
+      ("descriptor", `Assoc [ ("id", `String descriptor) ]);
+      ("message", `Assoc [ ("text", `String message) ]);
+      ("level", `String level);
+    ]
+
+let sarif_output hrules (cli_output : OutT.cli_output) =
+  let sarif_schema =
+    "https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/schemas/sarif-schema-2.1.0.json"
+  in
+  let engine_label =
+    match cli_output.OutT.engine_requested with
+    | Some `OSS
+    | None ->
+        "OSS"
+    | Some `PRO -> "PRO"
+  in
+  let run =
+    let hide_nudge =
+      (* TODO is_logged_in or is_pro or not is_using_registry *)
+      true
+    in
+    let rules = rules hide_nudge hrules in
+    let tool =
+      `Assoc
+        [
+          ( "driver",
+            `Assoc
+              [
+                ("name", `String (spf "Semgrep %s" engine_label));
+                ("semanticVersion", `String (*"%%VERSION%%"*) "1.56.0");
```

Comment:
here, the version is hardcoded. can we retrieve the version programmatically?

Reply:
Yes in Version.version

Reply:
done in [3bf4518](https://github.com/semgrep/semgrep/pull/9608/commits/3bf45181dd39089747601288efc06bae41cb1fe0)